From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mike <michaelgorney4@gmail.com>
Date: Mon, 8 Jun 2026 12:46:03 -0400
Subject: [PATCH 5/7] clang-mg: update traits


diff --git a/clang/lib/Parse/ParseDeclCXXMG.cpp b/clang/lib/Parse/ParseDeclCXXMG.cpp
index 52eff5ac6c93..42d80ab8bef3 100644
--- a/clang/lib/Parse/ParseDeclCXXMG.cpp
+++ b/clang/lib/Parse/ParseDeclCXXMG.cpp
@@ -4,6 +4,7 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include "clang/Basic/DiagnosticParse.h"
 #include "clang/Parse/Parser.h"
 #include "clang/AST/DeclCXX.h"
 #include "clang/AST/DeclTemplate.h"
-- 
2.54.0

