From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mike <michaelgorney4@gmail.com>
Date: Sun, 17 May 2026 20:01:13 -0400
Subject: [PATCH 1/7] clang-mg: update conditional-members


diff --git a/clang/lib/Parse/ParseDeclCXX.cpp b/clang/lib/Parse/ParseDeclCXX.cpp
index 877161b65b6c..f017824f0b9c 100644
--- a/clang/lib/Parse/ParseDeclCXX.cpp
+++ b/clang/lib/Parse/ParseDeclCXX.cpp
@@ -3490,6 +3490,15 @@ Parser::DeclGroupPtrTy Parser::ParseCXXClassMemberDeclarationWithPragmas(
   }
 }
 
+Parser::DeclGroupPtrTy Parser::ParseCXXMGConditionalMember(AccessSpecifier &AS,
+                              ParsedAttributes &AccessAttrs,
+                              DeclSpec::TST TagType, Decl *Tag) {
+  assert(getLangOpts().CPlusPlusMG &&
+         "ParseCXXMGConditionalMember should only be called in C++MG mode");
+  // TODO: Implement idk how yet
+  return nullptr;
+}
+
 void Parser::ParseCXXMemberSpecification(SourceLocation RecordLoc,
                                          SourceLocation AttrFixitLoc,
                                          ParsedAttributes &Attrs,
-- 
2.54.0

