From d7d4240042bd937e4e1d188c46b2365b5212f7c1 Mon Sep 17 00:00:00 2001
From: Mike <michaelgorney4@gmail.com>
Date: Sun, 17 May 2026 21:13:54 -0400
Subject: [PATCH 2/7] clang-mg: update conditional-members

---
 clang/lib/Parse/ParseDeclCXX.cpp | 41 +++++++++++++++++++++++++++++---
 1 file changed, 38 insertions(+), 3 deletions(-)

diff --git a/clang/lib/Parse/ParseDeclCXX.cpp b/clang/lib/Parse/ParseDeclCXX.cpp
index f017824f0b9c..da679e4712ad 100644
--- a/clang/lib/Parse/ParseDeclCXX.cpp
+++ b/clang/lib/Parse/ParseDeclCXX.cpp
@@ -3493,9 +3493,44 @@ Parser::DeclGroupPtrTy Parser::ParseCXXClassMemberDeclarationWithPragmas(
 Parser::DeclGroupPtrTy Parser::ParseCXXMGConditionalMember(AccessSpecifier &AS,
                               ParsedAttributes &AccessAttrs,
                               DeclSpec::TST TagType, Decl *Tag) {
-  assert(getLangOpts().CPlusPlusMG &&
-         "ParseCXXMGConditionalMember should only be called in C++MG mode");
-  // TODO: Implement idk how yet
+  assert(getLangOpts().CPlusPlusMG && "ParseCXXMGConditionalMember should only be called in C++MG mode");
+
+  // Consume "if" token & expect "constexpr" token
+  assert(Tok.is(tok::kw_if) && "Expected if token... how did we get here?");
+
+  auto ifLoc = Tok.getLocation();
+  ConsumeToken(); // consume the 'if'
+
+  // Expect 'constexpr' (or consteval) bc it just makes more sense man
+  if (!TryConsumeToken(tok::kw_constexpr) || !TryConsumeToken(tok::kw_consteval)) {
+    Diag(Tok.getLocation(), diag::err_expected);
+
+    // gpt says this will make the parser not get stuck
+    SkipUntil(tok::semi, tok::r_brace, StopBeforeMatch);
+  }
+
+  if (!TryConsumeToken(tok::l_paren)) {
+    Diag(Tok.getLocation(), diag::err_expected) << tok::l_paren;
+    SkipUntil(tok::semi, tok::r_brace, StopBeforeMatch);
+  }
+
+  auto cond = ParseExpression(); // Parse the condition expression
+
+  if (!TryConsumeToken(tok::r_paren)) {
+    Diag(Tok.getLocation(), diag::err_expected);
+    SkipUntil(tok::semi, tok::r_brace, StopBeforeMatch);
+  }
+
+  if (!TryConsumeToken(tok::l_brace)) {
+    Diag(Tok.getLocation(), diag::err_expected);
+    SkipUntil(tok::semi, tok::r_brace, StopBeforeMatch);
+  }
+
+  // TODO: Figure out what to do about ts
+  while (!TryConsumeToken(tok::r_brace)) {
+    ParseCXXClassMemberDeclarationWithPragmas(AS, AccessAttrs, TagType, Tag);
+  }
+
   return nullptr;
 }
 
-- 
2.54.0

