From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mike <michaelgorney4@gmail.com>
Date: Sun, 17 May 2026 23:44:50 -0400
Subject: [PATCH 4/7] clang-mg: update conditional-members


diff --git a/clang/include/clang/Parse/Parser.h b/clang/include/clang/Parse/Parser.h
index eae0d6791742..7e874e2c57af 100644
--- a/clang/include/clang/Parse/Parser.h
+++ b/clang/include/clang/Parse/Parser.h
@@ -3544,8 +3544,6 @@ private:
                                             ParsedAttributes &AccessAttrs,
                                             DeclSpec::TST TagType, Decl *Tag);
 
-<<<<<<< HEAD
-=======
   /// ParseCXXMGConditionalMember - Parse a C++MG 'if constexpr' member declaration
   DeclGroupPtrTy
   ParseCXXMGConditionalMember(AccessSpecifier &AS,
@@ -3558,7 +3556,6 @@ private:
                               ParsedAttributes &AccessAttrs,
                               DeclSpec::TST TagType, Decl *Tag);
 
->>>>>>> clang-mg: update conditional-members
   /// ParseConstructorInitializer - Parse a C++ constructor initializer,
   /// which explicitly initializes the members or base classes of a
   /// class (C++ [class.base.init]). For example, the three initializers
-- 
2.54.0

