From a95fb51385e822f3ccb3b94ffa6d3608fab8ce1b Mon Sep 17 00:00:00 2001
From: Mike <michaelgorney4@gmail.com>
Date: Sun, 17 May 2026 20:02:35 -0400
Subject: [PATCH] clang-mg: add language-option


diff --git a/clang/include/clang/Basic/LangOptions.def b/clang/include/clang/Basic/LangOptions.def
index 3bbb40ba09f0..0e26ff393017 100644
--- a/clang/include/clang/Basic/LangOptions.def
+++ b/clang/include/clang/Basic/LangOptions.def
@@ -57,6 +57,7 @@ LANGOPT(CPlusPlus17       , 1, 0, NotCompatible, "C++17")
 LANGOPT(CPlusPlus20       , 1, 0, NotCompatible, "C++20")
 LANGOPT(CPlusPlus23       , 1, 0, NotCompatible, "C++23")
 LANGOPT(CPlusPlus26       , 1, 0, NotCompatible, "C++26")
+LANGOPT(CPlusPlusMG       , 1, 1, NotCompatible, "C++MG")
 LANGOPT(ObjC              , 1, 0, NotCompatible, "Objective-C")
 LANGOPT(ObjCDefaultSynthProperties , 1, 0, Benign,
         "Objective-C auto-synthesized properties")
-- 
2.54.0

