From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mike <michaelgorney4@gmail.com>
Date: Sat, 23 May 2026 19:34:28 -0400
Subject: [PATCH] clang-mg: update traits

---
 clang/lib/Sema/SemaDeclCXXMG.cpp | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/clang/lib/Sema/SemaDeclCXXMG.cpp b/clang/lib/Sema/SemaDeclCXXMG.cpp
index 2f50d8cb4e3c..8061da693393 100644
--- a/clang/lib/Sema/SemaDeclCXXMG.cpp
+++ b/clang/lib/Sema/SemaDeclCXXMG.cpp
@@ -4,9 +4,23 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include "clang/Sema/SemaInternal.h"
+
+#include "clang/AST/DeclGroup.h"
+#include "clang/AST/DeclCXX.h"
+#include "clang/AST/DeclTemplate.h"
+#include "clang/AST/Expr.h"
+#include "clang/Sema/CXXFieldCollector.h"
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/Support/Casting.h"
+
+#include <cassert>
+#include <utility>
+
 using namespace clang;
 
-namespace {
 Decl *Sema::ActOnCXXMGTraitAssociatedTypeRequirement(
     Scope *S, Decl *TraitDecl, SourceLocation UsingLoc, SourceLocation NameLoc,
     IdentifierInfo *Name) {
@@ -69,5 +83,4 @@ void Sema::ActOnCXXMGTraitMemberDeclaration(Decl *TraitDecl,
     }
     Diag(D->getLocation(), diag::err_cxxmg_trait_invalid_member);
   }
-}
 }
\ No newline at end of file
-- 
2.54.0

